src/js/structs/__tests__/ServiceTree-test.js

```diff
@@ -147,26 +186,135 @@ describe('ServiceTree', function () {
     });
   });
 
+  describe('#filterItemsByFilter', function () {
+
+    beforeEach(function () {
+      this.instance = new ServiceTree({
```

Comment:
You're right there is a lot of repetition. But almost every object differs slightly. So it looks easier then it is.
For the sake of progress I like to refactor this in another PR.